Rename FrameLoaderClient to LocalFrameLoaderClient

achristensen07 opened the pull request:
#### f0bec4455223795c12327877045aa001d48a6806
<pre>
Rename FrameLoaderClient to LocalFrameLoaderClient
<a href="https://bugs.webkit.org/show_bug.cgi?id=256904">https://bugs.webkit.org/show_bug.cgi?id=256904</a>

Reviewed by Chris Dumez.

This is preparation for sharing decidePolicyForNavigationAction between LocalFrame and RemoteFrame.

* Source/WebCore/Modules/websockets/WebSocket.cpp:
* Source/WebCore/accessibility/AccessibilityObjectInterface.h:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
* Source/WebCore/bindings/js/ScriptController.cpp:
* Source/WebCore/bindings/js/ScriptControllerMac.mm:
* Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
* Source/WebCore/dom/DOMImplementation.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/DocumentStorageAccess.cpp:
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
* Source/WebCore/history/BackForwardCache.cpp:
* Source/WebCore/history/CachedFrame.cpp:
* Source/WebCore/history/CachedPage.cpp:
* Source/WebCore/html/HTMLAnchorElement.cpp:
* Source/WebCore/html/HTMLCanvasElement.cpp:
* Source/WebCore/html/HTMLFormElement.cpp:
* Source/WebCore/html/HTMLFrameSetElement.cpp:
* Source/WebCore/html/HTMLLinkElement.cpp:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/html/HTMLPlugInImageElement.cpp:
* Source/WebCore/html/ImageDocument.cpp:
* Source/WebCore/html/MediaDocument.cpp:
* Source/WebCore/html/ModelDocument.cpp:
* Source/WebCore/html/PluginDocument.cpp:
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
* Source/WebCore/loader/ContentFilter.cpp:
* Source/WebCore/loader/DocumentLoader.cpp:
* Source/WebCore/loader/DocumentThreadableLoader.cpp:
(WebCore::DocumentThreadableLoader::cancel):
* Source/WebCore/loader/DocumentWriter.cpp:
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::pageConfigurationWithEmptyClients):
* Source/WebCore/loader/EmptyFrameLoaderClient.h:
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::FrameLoader):
(WebCore::FrameLoader::loadInSameDocument):
(WebCore::LocalFrameLoaderClient::hasHTMLView const):
(WebCore::FrameLoaderClient::hasHTMLView const): Deleted.
* Source/WebCore/loader/FrameLoader.h:
* Source/WebCore/loader/HistoryController.cpp:
(WebCore::FrameLoader::HistoryController::restoreScrollPositionAndViewState):
* Source/WebCore/loader/LinkLoader.cpp:
* Source/WebCore/loader/LocalFrameLoaderClient.h: Renamed from Source/WebCore/loader/FrameLoaderClient.h.
* Source/WebCore/loader/MediaResourceLoader.cpp:
* Source/WebCore/loader/MixedContentChecker.cpp:
* Source/WebCore/loader/MixedContentChecker.h:
* Source/WebCore/loader/NetscapePlugInStreamLoader.cpp:
* Source/WebCore/loader/PingLoader.cpp:
* Source/WebCore/loader/PolicyChecker.cpp:
* Source/WebCore/loader/PolicyChecker.h:
* Source/WebCore/loader/ProgressTracker.cpp:
* Source/WebCore/loader/ResourceLoadNotifier.cpp:
(WebCore::ResourceLoadNotifier::didFailToLoad):
(WebCore::ResourceLoadNotifier::dispatchWillSendRequest):
(WebCore::ResourceLoadNotifier::dispatchDidReceiveResponse):
(WebCore::ResourceLoadNotifier::dispatchDidReceiveData):
(WebCore::ResourceLoadNotifier::dispatchDidFinishLoading):
(WebCore::ResourceLoadNotifier::dispatchDidFailLoading):
* Source/WebCore/loader/ResourceLoader.cpp:
* Source/WebCore/loader/SubframeLoader.cpp:
* Source/WebCore/loader/SubframeLoader.h:
* Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
* Source/WebCore/loader/cache/CachedImage.cpp:
* Source/WebCore/loader/cache/CachedResource.cpp:
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
* Source/WebCore/loader/icon/IconLoader.cpp:
* Source/WebCore/loader/ios/LegacyPreviewLoader.mm:
* Source/WebCore/loader/mac/ResourceLoaderMac.mm:
* Source/WebCore/page/Chrome.cpp:
* Source/WebCore/page/ContextMenuController.cpp:
* Source/WebCore/page/DOMWindowExtension.cpp:
* Source/WebCore/page/History.cpp:
* Source/WebCore/page/LocalDOMWindow.cpp:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::LocalFrame):
(WebCore::LocalFrame::createMainFrame):
(WebCore::LocalFrame::createSubframe):
(WebCore::LocalFrame::createSubframeHostedInAnotherProcess):
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/LocalFrameView.cpp:
* Source/WebCore/page/Navigator.cpp:
* Source/WebCore/page/Page.cpp:
(WebCore::createMainFrame):
* Source/WebCore/page/PageConfiguration.cpp:
(WebCore::PageConfiguration::PageConfiguration):
* Source/WebCore/page/PageConfiguration.h:
* Source/WebCore/page/PerformanceLogging.cpp:
* Source/WebCore/page/WebKitNamespace.cpp:
* Source/WebCore/platform/network/SynchronousLoaderClient.cpp:
(WebCore::SynchronousLoaderClient::shouldUseCredentialStorage):
(WebCore::SynchronousLoaderClient::canAuthenticateAgainstProtectionSpaceAsync):
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
* Source/WebCore/rendering/RenderLayer.cpp:
* Source/WebCore/style/StyleScopeRuleSets.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
* Source/WebCore/workers/service/ServiceWorkerGlobalScope.cpp:
* Source/WebCore/workers/service/ServiceWorkerProvider.cpp:
* Source/WebCore/workers/service/context/SWContextManager.cpp:
* Source/WebKit/WebProcess/Network/WebResourceLoader.cpp:
* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
* Source/WebKit/WebProcess/Storage/RemoteWorkerFrameLoaderClient.h:
(isType):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::installServiceWorker):
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.cpp:
(WebKit::WebSharedWorkerContextManagerConnection::launchSharedWorker):
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDidFailProvisionalLoad):
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.h:
(WebKit::toWebFrameLoaderClient):
* Source/WebKit/WebProcess/WebCoreSupport/WebResourceLoadObserver.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebFrameNetworkingContext.mm:
* Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp:
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::clientForMainFrame):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
* Source/WebKitLegacy/ios/WebView/WebPDFViewIOS.mm:
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameNetworkingContext.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(kit):
* Source/WebKitLegacy/mac/WebView/WebHTMLRepresentation.mm:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _commonInitializationWithFrameName:groupName:]):
(-[WebView initSimpleHTMLDocumentWithStyle:frame:preferences:groupName:]):

Canonical link: <a href="https://commits.webkit.org/264170@main">https://commits.webkit.org/264170@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f30303f9029595e5b83e6789f4cf55aceb7ec155

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6920 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7150 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7333 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8525 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/7165 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6925 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8407 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7089 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/10071 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7042 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/7712 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/6373 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8624 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/5092 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/6283 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14076 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/6778 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/6373 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/9206 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6852 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/5619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6228 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10415 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/797 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6609 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 54;
+	objectVersion = 51;
```
We should probably not be making this change.

achristensen07 committed:
Rename FrameLoaderClient to LocalFrameLoaderClient
https://bugs.webkit.org/show_bug.cgi?id=256904

Reviewed by Chris Dumez.

This is preparation for sharing decidePolicyForNavigationAction between LocalFrame and RemoteFrame.

* Source/WebCore/Modules/websockets/WebSocket.cpp:
* Source/WebCore/accessibility/AccessibilityObjectInterface.h:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
* Source/WebCore/bindings/js/ScriptController.cpp:
* Source/WebCore/bindings/js/ScriptControllerMac.mm:
* Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
* Source/WebCore/dom/DOMImplementation.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/DocumentStorageAccess.cpp:
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
* Source/WebCore/history/BackForwardCache.cpp:
* Source/WebCore/history/CachedFrame.cpp:
* Source/WebCore/history/CachedPage.cpp:
* Source/WebCore/html/HTMLAnchorElement.cpp:
* Source/WebCore/html/HTMLCanvasElement.cpp:
* Source/WebCore/html/HTMLFormElement.cpp:
* Source/WebCore/html/HTMLFrameSetElement.cpp:
* Source/WebCore/html/HTMLLinkElement.cpp:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/html/HTMLPlugInImageElement.cpp:
* Source/WebCore/html/ImageDocument.cpp:
* Source/WebCore/html/MediaDocument.cpp:
* Source/WebCore/html/ModelDocument.cpp:
* Source/WebCore/html/PluginDocument.cpp:
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
* Source/WebCore/html/parser/HTMLConstructionSite.cpp:
* Source/WebCore/loader/ContentFilter.cpp:
* Source/WebCore/loader/DocumentLoader.cpp:
* Source/WebCore/loader/DocumentThreadableLoader.cpp:
(WebCore::DocumentThreadableLoader::cancel):
* Source/WebCore/loader/DocumentWriter.cpp:
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::pageConfigurationWithEmptyClients):
* Source/WebCore/loader/EmptyFrameLoaderClient.h:
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::FrameLoader):
(WebCore::FrameLoader::loadInSameDocument):
(WebCore::LocalFrameLoaderClient::hasHTMLView const):
(WebCore::FrameLoaderClient::hasHTMLView const): Deleted.
* Source/WebCore/loader/FrameLoader.h:
* Source/WebCore/loader/HistoryController.cpp:
(WebCore::FrameLoader::HistoryController::restoreScrollPositionAndViewState):
* Source/WebCore/loader/LinkLoader.cpp:
* Source/WebCore/loader/LocalFrameLoaderClient.h: Renamed from Source/WebCore/loader/FrameLoaderClient.h.
* Source/WebCore/loader/MediaResourceLoader.cpp:
* Source/WebCore/loader/MixedContentChecker.cpp:
* Source/WebCore/loader/MixedContentChecker.h:
* Source/WebCore/loader/NetscapePlugInStreamLoader.cpp:
* Source/WebCore/loader/PingLoader.cpp:
* Source/WebCore/loader/PolicyChecker.cpp:
* Source/WebCore/loader/PolicyChecker.h:
* Source/WebCore/loader/ProgressTracker.cpp:
* Source/WebCore/loader/ResourceLoadNotifier.cpp:
(WebCore::ResourceLoadNotifier::didFailToLoad):
(WebCore::ResourceLoadNotifier::dispatchWillSendRequest):
(WebCore::ResourceLoadNotifier::dispatchDidReceiveResponse):
(WebCore::ResourceLoadNotifier::dispatchDidReceiveData):
(WebCore::ResourceLoadNotifier::dispatchDidFinishLoading):
(WebCore::ResourceLoadNotifier::dispatchDidFailLoading):
* Source/WebCore/loader/ResourceLoader.cpp:
* Source/WebCore/loader/SubframeLoader.cpp:
* Source/WebCore/loader/SubframeLoader.h:
* Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
* Source/WebCore/loader/appcache/ApplicationCacheHost.cpp:
* Source/WebCore/loader/cache/CachedImage.cpp:
* Source/WebCore/loader/cache/CachedResource.cpp:
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
* Source/WebCore/loader/icon/IconLoader.cpp:
* Source/WebCore/loader/ios/LegacyPreviewLoader.mm:
* Source/WebCore/loader/mac/ResourceLoaderMac.mm:
* Source/WebCore/page/Chrome.cpp:
* Source/WebCore/page/ContextMenuController.cpp:
* Source/WebCore/page/DOMWindowExtension.cpp:
* Source/WebCore/page/History.cpp:
* Source/WebCore/page/LocalDOMWindow.cpp:
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::LocalFrame):
(WebCore::LocalFrame::createMainFrame):
(WebCore::LocalFrame::createSubframe):
(WebCore::LocalFrame::createSubframeHostedInAnotherProcess):
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/LocalFrameView.cpp:
* Source/WebCore/page/Navigator.cpp:
* Source/WebCore/page/Page.cpp:
(WebCore::createMainFrame):
* Source/WebCore/page/PageConfiguration.cpp:
(WebCore::PageConfiguration::PageConfiguration):
* Source/WebCore/page/PageConfiguration.h:
* Source/WebCore/page/PerformanceLogging.cpp:
* Source/WebCore/page/WebKitNamespace.cpp:
* Source/WebCore/platform/network/SynchronousLoaderClient.cpp:
(WebCore::SynchronousLoaderClient::shouldUseCredentialStorage):
(WebCore::SynchronousLoaderClient::canAuthenticateAgainstProtectionSpaceAsync):
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
* Source/WebCore/rendering/RenderLayer.cpp:
* Source/WebCore/style/StyleScopeRuleSets.cpp:
* Source/WebCore/workers/service/ServiceWorkerContainer.cpp:
* Source/WebCore/workers/service/ServiceWorkerGlobalScope.cpp:
* Source/WebCore/workers/service/ServiceWorkerProvider.cpp:
* Source/WebCore/workers/service/context/SWContextManager.cpp:
* Source/WebKit/WebProcess/Network/WebResourceLoader.cpp:
* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
* Source/WebKit/WebProcess/Storage/RemoteWorkerFrameLoaderClient.h:
(isType):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::installServiceWorker):
* Source/WebKit/WebProcess/Storage/WebSharedWorkerContextManagerConnection.cpp:
(WebKit::WebSharedWorkerContextManagerConnection::launchSharedWorker):
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDidFailProvisionalLoad):
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.h:
(WebKit::toWebFrameLoaderClient):
* Source/WebKit/WebProcess/WebCoreSupport/WebResourceLoadObserver.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebFrameNetworkingContext.mm:
* Source/WebKit/WebProcess/WebPage/WebCookieJar.cpp:
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::clientForMainFrame):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
* Source/WebKitLegacy/ios/WebView/WebPDFViewIOS.mm:
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameNetworkingContext.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(kit):
* Source/WebKitLegacy/mac/WebView/WebHTMLRepresentation.mm:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _commonInitializationWithFrameName:groupName:]):
(-[WebView initSimpleHTMLDocumentWithStyle:frame:preferences:groupName:]):

Canonical link: https://commits.webkit.org/264170@main
